nsjail.cc

```diff
@@ -115,6 +122,70 @@ static bool setTimer(nsjconf_t* nsjconf) {
 	return true;
 }
 
+static bool pipeTraffic(nsjconf_t* nsjconf, int listenfd) {
+	std::vector<struct pollfd> fds;
+	fds.reserve(nsjconf->pipes.size() * 2 + 1);
+	for (const auto& p : nsjconf->pipes) {
+		fds.push_back({
+		    .fd = p.first,
+		    .events = POLLIN,
+		    .revents = 0,
+		});
+		fds.push_back({
+		    .fd = p.second,
+		    .events = POLLOUT,
+		    .revents = 0,
+		});
+	}
+	fds.push_back({
+	    .fd = listenfd,
+	    .events = POLLIN,
+	    .revents = 0,
+	});
+	LOG_D("Waiting for fd activity");
+	while (poll(fds.data(), fds.size(), -1) > 0) {
+		if (fds.back().revents != 0) {
+			LOG_D("New connection ready");
```

Comment:
Since its `LOG_D` maybe it should print out the particular fd. Though that is not that crucial.

Reply:
It's always the listening socket.
So I think the message is enough.